packages/dd-trace/src/exporters/common/request.js

```diff
@@ -17,7 +18,9 @@ const containerId = docker.id()
 
 let activeRequests = 0
 
-function request (data, options, keepAlive, callback) {
+const RECOVERABLE_HTTP_STATUS = [408]
```

Comment:
this is a time out, which can be safely retried

Reply:
I'd say that's the recommended way to deal with 408: 
https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.7 
> If the client has an outstanding request in transit, the
   client MAY repeat that request on a new connection.

https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines
> The HTTP status code returned from the service can help to indicate whether the failure is transient. You may need to examine the exceptions generated by a client or the retry framework to access the status code or to determine the equivalent exception type. The following HTTP codes typically indicate that a retry is appropriate: 
408 Request Timeout

Reply:
500s could be retried too indeed

Reply:
It _can_ be retried sure, but I'm arguing that I don't think we should since we just risk overloading an already overloaded system on either side on the connection. What other use cases do you have in mind for timeouts?

Reply:
I removed the retry logic. If we need something at CI Vis, I'll do it at the writer